mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -258,6 +290,27 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.0.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
```

Comment:
I think that's the user's problem if they overwrite k or other parameters after setting the initial model.  However, I'm Ok with addressing this in the follow-up PR for overriding the Params when initialModel is set.